model/database/orm.py

```diff
@@ -0,0 +1,100 @@
+
+
+class DjangoORM:
+    """
+    Encapsulate Django setup and ORM access in a class
+    This should normally be used like the following:
+    orm = DjangoORM
+    orm.connect()
+    records = orm.data_model.<TableName>.objects.<Filter/OrderBy/..>
+    """
+
+    data_model = None
+    variable_model = None
+
+    @staticmethod
+    def add_webapp_path():
+        """
+        Get path to the autoreduce_webapp and add this path to the system path
+        This will enable importing of the autoreduce_webapp (and it's models)
+        """
+        path = os.path.join(get_project_root(), 'WebApp', 'autoreduce_webapp')
+        if path not in sys.path:
+            # Can't properly replicate this in a testing environment due to travis
+            # constraint on sys path access and mocking would break underlying dependency
+            # but it's simple enough we don't need to worry about testing it
+            sys.path.append(path)  # pragma: no cover
+
+    @staticmethod
+    def setup_django():
+        """
+        Use the WebApp settings to initialise a django instance that can be used for model access
+        """
+        try:
+            # import here to avoid failure without calling add_webapp_path first
+            # pylint:disable=import-outside-toplevel
+            from WebApp.autoreduce_webapp.autoreduce_webapp.settings import DATABASES, ORM_INSTALL
+            settings.configure(
+                DATABASES=DATABASES,
+                INSTALLED_APPS=ORM_INSTALL,
+            )
+            django.setup()
+        except RuntimeError as excep:
+            logging.warning("Exception raised when attempting to setup: %s", excep)
+
+    def _get_data_model(self):
+        """
+        Importing the ORM objects for reduction data
+        """
+        if not self.data_model:
+            # pylint:disable=import-outside-toplevel,import-error
+            import reduction_viewer.models as data_model
+            self.data_model = data_model
+        return self.data_model
+
+    def _get_variable_model(self):
+        """
+        Import the ORM objects for reduction variables
+        """
+        if not self.variable_model:
+            # pylint:disable=import-outside-toplevel,import-error
+            import reduction_variables.models as variable_model
+            self.variable_model = variable_model
+        return self.variable_model
+
+    def connect(self):
+        """
+        Control function to actually perform the setup of the django ORM connection
+        :return: (bool) was the setup successful?
+        """
+        self.add_webapp_path()
+        self.setup_django()
+        try:
+            if not self.data_model:
+                self._get_data_model()
+            self.data_model.Instrument.objects.first()
+            if not self.variable_model:
+                self._get_variable_model()
+            self.variable_model.Variable.objects.first()
+        # pylint:disable=bare-except
+        except:
```

Comment:
Would a comment to explain why you've used a _bare except_ be useful here for the benefit of future devs